Demo/Demo/Features/CardDemoViewController.swift

```diff
@@ -0,0 +1,105 @@
+        view.distribution = .fillEqually
+        view.spacing = Constants.stackViewSpacing
+        return view
+    }()
+
+    lazy var cardNumberTextField: UITextField = {
+        let textField = textField(placeholder: "Card Number")
+        textField.keyboardType = .numberPad
+        return textField
+    }()
+
+    lazy var expirationTextField: UITextField = {
+        let textField = textField(placeholder: "Expiration")
+        textField.keyboardType = .numberPad
+        return textField
+    }()
+
+    lazy var cvvTextField: UITextField = {
+        let textField = textField(placeholder: "CVV")
+        textField.keyboardType = .numberPad
+        return textField
+    }()
+
+    lazy var payButton: UIButton = {
+        let payButton = button(title: "Pay", action: #selector(didTapPayButton))
+        payButton.layer.cornerRadius = 8
+        payButton.backgroundColor = .systemBlue
+        payButton.tintColor = .white
+        return payButton
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.backgroundColor = .systemBackground
+
+        processOrder(paymentMethod: .card, state: .notStarted)
+        setupUI()
+        setupConstraints()
+    }
+
+    func setupUI() {
+        view.addSubview(mainStackView)
+
+        mainStackView.addArrangedSubview(cardNumberTextField)
+        mainStackView.addArrangedSubview(secondaryStackView)
+        mainStackView.addArrangedSubview(payButton)
+
+        secondaryStackView.addArrangedSubview(expirationTextField)
+        secondaryStackView.addArrangedSubview(cvvTextField)
+    }
+
+    func setupConstraints() {
+        NSLayoutConstraint.activate([
+            mainStackView.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            mainStackView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+
+            secondaryStackView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 16),
+            secondaryStackView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -16),
+
+            cardNumberTextField.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 16),
+            cardNumberTextField.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -16),
+            cardNumberTextField.heightAnchor.constraint(equalToConstant: 40),
+
+            expirationTextField.heightAnchor.constraint(equalToConstant: 40),
+            cvvTextField.heightAnchor.constraint(equalToConstant: 40),
+
+            payButton.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 16),
+            payButton.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -16),
+            payButton.heightAnchor.constraint(equalToConstant: 40)
+        ])
+    }
+
+    @objc func didTapPayButton() {
+        // TODO: Pass card / order details into fetchOrderID
+        processOrder(paymentMethod: .card, state: .orderProcessing)
+        fetchOrderID(paymentMethod: .card)
```

Comment:
Fetch orderID should happen before they click the button, when the view loads

Reply:
Ope, this was my misunderstanding! Will get  this updated.